frontend/compute-system/src/components/AppStore/AppDetails/components/CommentFom.tsx

```diff
@@ -0,0 +1,63 @@
+import { TextField, Container, Button, Grid } from "@material-ui/core";
+import * as React from "react";
+import Stars from "../../AppRating/components/Stars";
+import commentFormStyles from "./CommentFormStyles";
+
+interface CommentFormProps {
+  id: string;
+}
+
+const CommentForm = (props: CommentFormProps) => {
+  const classes = commentFormStyles();
+
+  const [rate, setRate] = React.useState(10);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    var value = Number(event.target.value);
+    if (value <= 10 && value >= 0) setRate(value);
+  };
+
+  return (
+    <Container>
+      <form>
+        <Grid
+          item
+          container
+          alignItems="center"
+          spacing={2}
+          className={classes.formContainer}
+        >
+          <Grid item xs={2}>
+            <TextField
+              required
+              label="Rate"
+              type="number"
+              value={rate}
+              className={classes.number}
+              onChange={handleChange}
+            />
+          </Grid>
+          <Grid item xs={10}>
+            <Stars rate={rate} />
+          </Grid>
+          <Grid item xs={12}>
+            <TextField
```

Comment:
validate input length (max 5000 chars) - best in a controlled component
what would be the best is to color the input border for example red and next to add button write sth like `Your comment is too long` and block the add button